### DISCUSSION TITLE
The Future: It's So Bright It Burns the Eyes

### DISCUSSION THREAD

**Discussion by leycec** [beartype/beartype#292] [Announcements]
Welcome to @beartype's official roadmap for 2023—*early* 2024. The road ahead is patchy, filled with potholes, and probably not great for your suspension. This is that road:

1. **October, 2023:** Officially support the recently released **Python 3.12.** Prolly should have done something about that this summer, huh? :face_exhaling: 
1. **November, 2023:** *Finally* finish documenting our new `beartype.claw` and `beartype.peps` subpackages at [our ReadTheDocs (RTD)-hosted site](https://beartype.readthedocs.io/en/latest).
1. **December, 2023:** Deeply type-check set type hints (e.g., `set[...]`, `collections.abc.Set[...]`, `typing.AbstractSet[...]`). Calling @langfield. If you even use Python anymore, <sup>*curse you, Haskell!*</sup> @beartype can help. If not, @beartype can bring you back.
1. **January, 2024:** Deeply type-check mapping type hints (e.g., `dict[...]`, `collections.abc.Mapping[...]`, `typing.MutableMapping[...]`). Calling @langfield even more.
1. **February, 2024:** Probably [Plum](https://github.com/beartype/plum/)-related DOOR stuff like [`beartype.door.TypeHint` support for **NumPy type hints**](https://github.com/beartype/beartype/issues/212). Calling @wesselb. This is your moment.
1. **March, 2024:** Deeply type-check generic container type hints (e.g., `collections.abc.Iterable[...]`, `collections.abc.Collection[...]`). @langfield intensifies.
1. **April—_gods knows how long_, 2024:** [`TypeVar`-based type-stable parameter checking](https://github.com/beartype/beartype/issues/236)! Finally! At last! @beartype does something you want!

**tl;dr:** It's gonna be awhile. See everybody in 2024. `</sigh>`

**Comment by PhilipVinc** [comment]
How to you plan to parse TypeVar stuff? Do you have a vague plan? Because it will require a sizeable change from current's beartype approach of verifying one argument at a time and you'll instead need to resolve them all at once, or am I mistaken?

**Comment by leycec** [comment]
**Ha-ah!** It's my favourite numerical physicist, back with the hard-hitting questions that sting like a bee, float like a butterfly, and bite like a python. Let's [take this to #236, where the painful Truth shall at last be revealed](https://github.com/beartype/beartype/issues/236#issuecomment-1767674236). :open_mouth:
